src/SFML/Main/CMakeLists.txt

```diff
@@ -15,6 +15,7 @@ endif()
 
 # define the sfml-main target
 sfml_add_library(sfml-main STATIC SOURCES ${SRC})
+set_target_properties(sfml-main PROPERTIES POSITION_INDEPENDENT_CODE ON)
```

Comment:
I think this problem applies to all non-DLL platforms that may use this library. Sure I can add a comment.

Reply:
I am not very familiar with Android development but, if I understand correctly, the issue happens on Android only, because native code has to be in a dynamic library. There is no main native executable. So the sfml-main static library gets linked into a shared library, which needs position independent code.
On other platforms, the sfml-main static library can be linked into the main executable directly so PIC is not needed.
Or did I miss something?

Reply:
Basically this, yes. The other platforms will link the static library into an executable where you'd want position dependent code (for performance reasons). Newer compilers have an alternative that works for executables (PIE instead of PIC), but CMake should enable it on its own.

Reply:
> This property does nothing on DLL platforms so I see no reason to try to avoid setting it for Windows builds.
This might slow down the included code a tiny bit, but considering it's just a wrapper around `main()`, should be fine.

Reply:
> the issue happens on Android only, because native code has to be in a dynamic library

Ah, this explains why we haven't caught this bug earlier. That makes sense because these issues only arise when you link to a shared library which itself links to a static. It takes a few layers of indirection for this to cause an issue.

Because this seems to be a problem that only affects Android I won't make other platforms include this flag.